simulation/pathfinder/main.cc

```diff
@@ -1,19 +1,81 @@
 #include "Position.hpp"
 #include "Queue.hpp"
 #include <vector>
+#include "stdio.hpp"
+#include <deque>
 
-std::vector<Position> pathPos = {Position{12,0}, Position{11,0}, Position{10,0},
-                        Position{9,0}, Position{8,0}, Position{7,0}, Position{7,1},
-                        Position{7,2}, Position{7,3}, Position{7,4}, Position{7,5},
-                        Position{7,6}};
+std::vector<Position> pathPos = {
+    Position{6,3}, Position{7,3}, Position{8,3}, Position{9,3}, Position{9,4}, Position{8,4}, Position{7,4}, Position{7,5}, Position{7,6}
+};
 
 Queue<Position> queue;
 
+//enum {F , B , L, R} direction;
+
+//Queue<char> path;
+//std::deque<char> path
+std::deque<char>path;
+
 int main(void)
 {
+    int init_row= NULL,init_col = NULL;
+    int final_col,final_row ;
+
     queue.init();
-    for (int i = 0; i < pathPos.size(); i++) {
+    for (int i = 0; i < pathPos.size(); i++) 
+    {
         queue.pushToBack(pathPos[i]);
     }
+
+    for(int i=0;i < pathPos.size(); i++)
+    {
+        if(init_col == NULL && init_row == NULL)
+        {
+            init_row = queue.peekFromFront().row;
+            init_col = queue.popFromFront().col;
+
+            final_col = queue.peekFromFront().col;
+            final_row = queue.popFromFront().row;
+        }
+        else
+        {
+            init_row = final_row;
+            init_col = final_col;
+
+            final_col = queue.peekFromFront().col;
+            final_row = queue.popFromFront().row;
+        }
+
+
+        if(init_row == (final_row-1))
+                path.push_back('F');
+        else if(init_row == (final_row+1) )
+                path.push_back('B');
```

Comment:
Sorry I didn't make it clear but our micromouse (and absolutely most micromouse) won't have 'going back' operation. Please use make a logic without 'B'

Reply:
As i wrote at the comment we need backward moving status in order to distinguish whether robot is facing down or up.
Basically moving status is same.